tests/nfs/nfs_ha_failover_delete_export.py

```diff
@@ -0,0 +1,107 @@
+from threading import Thread
+from time import sleep
+
+from nfs_operations import cleanup_cluster, perform_failover, setup_nfs_cluster
+
+from cli.ceph.ceph import Ceph
+from cli.exceptions import ConfigError, OperationFailedError
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def delete_export(clients, nfs_name, nfs_export):
+    for i in range(len(clients)):
+        try:
+            Ceph(clients[0]).nfs.export.delete(nfs_name, f"{nfs_export}_{i}")
+        except Exception:
+            raise OperationFailedError(f"Fail to delete {nfs_export}_{i}")
+
+
+def run(ceph_cluster, **kw):
+    """Perform failover when linux untar is running
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+    """
+    config = kw.get("config")
+    nfs_nodes = ceph_cluster.get_nodes("nfs")
+    clients = ceph_cluster.get_nodes("client")
+    port = config.get("port", "2049")
+    version = config.get("nfs_version", "4.1")
+    no_clients = int(config.get("clients", "1"))
+    no_servers = int(config.get("servers", "3"))
+    ha = bool(config.get("ha", False))
+    vip = config.get("vip", None)
+
+    # If the setup doesn't have required number of clients, exit.
+    if no_clients > len(clients):
+        raise ConfigError("The test requires more clients than available")
+
+    # If the setup doesn't have required number of nfs server, exit.
+    if no_servers > len(nfs_nodes):
+        raise ConfigError("The test requires more servers than available")
+
+    clients = clients[:no_clients]
+    servers = nfs_nodes[:no_servers]
+    fs_name = "cephfs"
+    nfs_name = "cephfs-nfs"
+    nfs_export = "/export"
+    nfs_mount = "/mnt/nfs"
+    fs = "cephfs"
+    nfs_server_name = [nfs_node.hostname for nfs_node in servers]
+
+    try:
+        # Setup nfs cluster
+        setup_nfs_cluster(
+            clients,
+            nfs_server_name,
+            port,
+            version,
+            nfs_name,
+            nfs_mount,
+            fs_name,
+            nfs_export,
+            fs,
+            ha,
+            vip,
+        )
+
+        sleep(3)
+
+        # Delete all exports
+        th = Thread(
+            target=delete_export,
+            args=(
+                clients,
+                nfs_name,
+                nfs_export,
+            ),
+        )
+        th.start()
+
+        # Perfrom and verify failover
```

Comment:
Deleting 1 export will happen quickly. Recommeneded way to test this scenario will be to create multiple exports and then perform failover. Deleting the export should run until the node is back online. Later check no stale export entry is present.

Reply:
Updated.

Log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-WACQRG/Nfs-Ganesha_perform_failover_and_delete_the_export,_check_the_export_list_0.log